### DISCUSSION TITLE
Spot small images?

### DISCUSSION THREAD

**Discussion by Elaphe-GitHub** [ImageMagick/ImageMagick#4179] [General]
Hi. I have large collection of images in a set of folders. We're talking about thousands of files. I would like to spot images with small dimensions. What would be the fastest way to do this? I suppose Imagemagick can do it. What would be the fastest command for this? Any other program or way to do it? Once Imagemagick reports the dimensions of an image, I could pass that information to a script to copy or not the file depending on a condition. With all the small images together, my idea is to use a tool such as Waifu2X to smart upscale the images without loosing definition. Oh, is Imagemagick also capable of doing this? Thank you!!!

**Comment by snibgo** [comment]
What version of IM, on what platform?

Yes, IM can tell you the dimensions of an image. For example:
```
magick toes.png -format "WW=%w\nHH=%h\n" info:
```

The output is:
```
WW=267
HH=233
```
So you can have a shell script that loops through files, calls "magick" for each file, and sets environment variables WW and HH. Your script can then take actions according to the values.
